src/Registry.php

```diff
@@ -79,10 +83,27 @@ private function resetOrClearManager(string $managerName, string $serviceId): vo
 
         assert($manager instanceof EntityManagerInterface);
 
-        if ((! $manager instanceof LazyLoadingInterface && ! $manager instanceof LazyObjectInterface) || $manager->isOpen()) {
-            $manager->clear();
+        // Determine if the version of symfony/dependency-injection is >= 7.3
+        /** @phpstan-ignore function.alreadyNarrowedType */
+        $sfNativeLazyObjects = method_exists('Symfony\Component\DependencyInjection\ContainerBuilder', 'findTaggedResourceIds');
 
-            return;
+        if (PHP_VERSION_ID < 80400 || ! $sfNativeLazyObjects) {
+            if ((! $manager instanceof LazyLoadingInterface && ! $manager instanceof LazyObjectInterface) || $manager->isOpen()) {
+                $manager->clear();
+
+                return;
+            }
+        } else {
+            $r = new ReflectionClass($manager);
+            if ($r->isUninitializedLazyObject($manager)) {
+                return;
```

Comment:
Doesn't this change the intended flow? Previous logic was calling $managed->clear() regardless if proxy is lazy or not, as long as $manager->isOpen() returns true

Reply:
There is no reason to call anything on the object if it's initializer function was not even called yet, i.e. if the lazy object was not yet initialized and we remove this check, then the manager will get initialized just to check if it's open. That's why i think it's safe to leave it as uninitialized. It's what the "symfony-bridge" is already doing.